src/probnum/problems/_problems.py

```diff
@@ -1 +1,204 @@
-"""Definitions of problems solved by probabilistic numerical methods."""
+"""Definitions of problems currently solved by probabilistic numerical methods.
+
+If a problem is not listed here, we have not implemented a probabilistic
+numerical method for it yet.
+"""
+
+import dataclasses
+import typing
+
+import numpy as np
+import scipy.sparse
+
+import probnum.filtsmooth as pnfs
+import probnum.linops as pnlo
+import probnum.random_variables as pnrv
+import probnum.type as pntp
+
+
+@dataclasses.dataclass
+class RegressionProblem:
+    r"""Regression problem.
+
+    Fit a stochastic process to data, given a likelihood (realised by a :obj:`DiscreteGaussian` transition).
+    Solved by Kalman filtering and smoothing in :mod:`probnum.filtsmooth`.
+
+    Parameters
+    ----------
+    observations
+        Observations of the latent process.
+    locations
+        Grid-points on which the observations were taken.
+    likelihood
+        Likelihood of the observations; that is, relation between the latent process and the observed values.
+        Encodes for example noise.
+    solution
+        Closed form, analytic solution to the problem. Used for testing and benchmarking.
+
+    Examples
+    --------
+    >>> obs = [11.4123, -15.5123]
+    >>> loc = [0.1, 0.2]
+    >>> rp = RegressionProblem(observations=obs, locations=loc)
+    >>> rp
+    RegressionProblem(observations=[11.4123, -15.5123], locations=[0.1, 0.2], likelihood=None, solution=None)
+    >>> rp.observations
+    [11.4123, -15.5123]
+    """
+
+    observations: np.ndarray
+    locations: np.ndarray
+
+    # Optional, because it should be specifiable without explicit likelihood info.
+    likelihood: typing.Optional[pnfs.statespace.DiscreteGaussian] = None
+
+    # For testing and benchmarking
+    solution: typing.Optional[
+        typing.Callable[[np.ndarray], typing.Union[float, np.ndarray]]
+    ] = None
+
+
+@dataclasses.dataclass
+class IVProblem:
```

Comment:
Wouldn't it make more sense to write this out `InitialValueProblem`? I feel like this would be more in line with the other problem classes, but also I don't have a strong opinion.

Reply:
Turns out that both nathanael and I agree with you :) Thanks for the comment!